docs/core/vpn.md

```diff
@@ -7,22 +7,109 @@ parent: Core
 
 # VPN
 
-Each node is connected to the leader using [WireGuard](https://www.wireguard.com/) in a star network topology.
+Each node is connected to the leader using
+[WireGuard](https://www.wireguard.com/) in a star network topology, where
+the leader node is the hub of the VPN links so it is connected to any
+other node.
 
 The VPN has its own network, the default is `10.5.4.0/24` and it listens on port 55820.
 Make sure this address is not already used inside your existing network and the UDP port 55820 is open on your firewall.
 You can change both configurations during cluster initialization.
 
-To inspect the VPN status you can use both `systemctl status wg-quick@wg0.service` and `wg show` commands.
-When the VPN is correctly configured, the leader now always takes the first IP for the above CIDR.
+## Promote a worker to leader
+
+Changing the cluster leader implies all VPN connections must be
+reconfigured to point to the new VPN hub. The future leader node must
+provide a VPN endpoint that is reachable by any other node. For this
+reason, **be prepared in advance** to deal with a leader failure by
+configuring a reachable endpoint address on one or more worker nodes!
+
+For instance if node with ID `3` has a public and static IP address
+`1.2.3.4` log on the leader node and run the following command:
+
+    redis-cli HSET node/3/vpn endpoint 1.2.3.4:55820
+
+A DNS host name can be set too, if there is one available:
+
+    redis-cli HSET node/3/vpn endpoint node3.example.com:55820
+
```

Comment:
I'm not sure we should advise the user about a manual configuration.
If this is required, its should be automatic or forced somehow.